### PR TITLE
#170947492 user is able to signup

### PR DESCRIPTION
**Pull request by Ntare22**
#### What does this PR do?
create user sign up, navbar, and footer
#### Description of Task to be completed?
- create files for navbar, footer, and signup components
- connect css to storybook
- create sample storybook stories for signup and home
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[ #170947492](https://www.pivotaltracker.com/story/show/170947492)
#### Questions:

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { render } from 'react-dom';
 import { Provider } from 'react-redux';
 import App from './components/App';
 import store from './store';
+// import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (izzett222) [inline src/index.js]: could you remove the comments

File: src/redux/reducers/authReducer.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const authReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SIGNUP_SUCCESS:
+      return {
+        ...state, user: action.payload, isLoggedIn: true,
+      };
+    case SIGNUP_ERROR:
+      return {
+        ...state, error: action.payload,
+      };
```
Comment (izzett222) [inline src/redux/reducers/authReducer.js]: isn't it safer to turn back user and isLoggedIn to null when there is an error

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: src/redux/reducers/authReducer.js (hunk 2 above)
Comment (Ntare22) [reply src/redux/reducers/authReducer.js]: done!!

File: src/index.js (hunk 1 above)
Comment (Ntare22) [reply src/index.js]: done!

File: src/__tests__/App.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 
 const wrapper = shallow(<login />);
 
-describe("Sample components tests", () => {
-  it("Should Render login Component", () => {
+describe('Sample components tests', () => {
+  it('Should Render login Component', () => {
     expect(wrapper.exists()).toBe(true);
   });
 });
```
Comment (Baraka-Mugisha) [inline src/__tests__/App.js]: Do we need this file anymore?

File: src/components/sharedComponents/APICall.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  };
+  return (
+    <div>
+      <button type="button" onClick={handleApiCall}>Click</button>
+      <h3>{message.message}</h3>
+    </div>
+  );
+};
+
+export default ApiCall;
```
Comment (bbaime98) [inline src/components/sharedComponents/APICall.js:20]: we don't need this file any more, consider removing it.

**Review by bbaime98 — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

File: src/components/authentication/Signup.test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+});
+
+it('should render form item component', () => {
+  expect(formItemComponent.exists()).toBe(true);
+});
+
+describe('Signup component tests', () => {
+  it('should render signup title', () => {
+    expect(toJson(signupComponent.dive())).toMatchSnapshot();
+  });
```
Comment (ceelogre) [inline src/components/authentication/Signup.test.js]: These tests are shallow. Find a way to test the actual data.

File: src/components/authentication/Signup.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+import '../../scss/signup.scss';
+import BackgroundImage from '../../assets/backg.jpg';
+import FormItem from './FormItem';
+
+
+const Signup = (props) => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```
Comment (ceelogre) [inline src/components/authentication/Signup.js:26]: This file is too big! I'm finding it hard to follow thru the flow, please consider moving certain parts like useState hook variable to external utils and refactoring if-statements.

File: src/redux/actions/signupAction.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+/* eslint-disable import/prefer-default-export */
+import axios from 'axios';
+import { SIGNUP_ERROR, SIGNUP_SUCCESS } from './actionsTypes';
+
+
+export const signupAction = (user) => async (dispatch) => {
+  try {
+    const res = await axios({
+      method: 'post',
+      url: 'https://devrepublic-bn-backend.herokuapp.com/api/v1/auth/register',
```
Comment (ceelogre) [inline src/redux/actions/signupAction.js]: Is this URI still valid since you changed Orgs?

File: src/redux/reducers/signupReducer.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      return {
+        ...state, user: { token: action.payload.token }, isLoggedIn: true,
+      };
+    case SIGNUP_ERROR:
+      return {
+        ...state, error: action.payload, isLoggedIn: null,
+      };
+    default:
+      return state;
+  }
```
Comment (ceelogre) [inline src/redux/reducers/signupReducer.js]: Consider my communication about switch statements.

**Review by ceelogre — CHANGES_REQUESTED** [review]

File: src/redux/actions/signupAction.js (hunk 7 above)
Comment (Ntare22) [reply src/redux/actions/signupAction.js]: Yes

**Review by ceelogre — COMMENTED** [review]
Create a new review app.

**Comment by bbaime98** [comment]
image
- Can you enable the register button only if when the password meets the requirements?

**Comment by ceelogre** [comment]
Capture d’écran 2020-05-13 à 14 45 34

I'm stuck at the login screen if I try to signup with those details.

**Review by jabichris — CHANGES_REQUESTED** [review]
Nice work @Ntare22. Though everything is working as intended, this PR needs a bit more work so that it can be merged.
* We agreed as a team to not introduce disabled new eslint rules, If we know why they are used for in the first place.
* Why are you introducing a new empty file? `src/redux/actions/index.js`
* Verify and remove all inline commented code
* Take those errors that are coming if I use one character name, display them for a good user experience(refer to the screenshot)
Screen Shot 1399-02-24 at 3 22 26 PM

* Consider repositioning the signup form. As you can see in the screenshot it doesn't look good.
Screen Shot 1399-02-24 at 3 20 00 PM

* Consider doing other small changes that I didn't mention(like writing down `how this should be manually tested`) to your task so that it can be merged soon

**Comment by jabichris** [comment]
> Capture d’écran 2020-05-13 à 14 45 34
> 
> I'm stuck at the login screen if I try to signup with those details.

Screen Shot 1399-02-24 at 3 22 26 PM
The above is the issue. @Ntare22 should fix this

**Comment by Ntare22** [comment]
> image
> 
> * Can you enable the register button only if when the password meets the requirements?

Hi @bienaime, I manage to do some thing similar that notifies the user that they have entered the wrong password

**Review by ceelogre — APPROVED** [review]
